### PR TITLE
Add homework 02 task 05 and test 05

### PR DESCRIPTION
**Pull request by mukhaw**

### CODE REVIEW

File: hm_1/tasks/task_02_05.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+def detect_index(some_iterable_object, position):
+    if isinstance(some_iterable_object, dict):
+        if position in list(some_iterable_object.keys()):
+            return list(some_iterable_object.keys()).index(position)
+        else:
+            return list(some_iterable_object.values()).index(position)
+    return some_iterable_object.index(position)
+
+
+def custom_range(some_iterable_object, start="", stop="", step=1):
```
Comment (Cjkjvfnby) [inline hm_1/tasks/task_02_05.py:10]: Can some_iterable_object have an empty string as a value?

File: test/test_02_05.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+
+def test_custom_range_string_with_start_stop_returns_part_of_string():
+    actual_result = custom_range(string.ascii_lowercase, "g", "p")
+    expected_result = ["g", "h", "i", "j", "k", "l", "m", "n", "o"]
+    assert actual_result == expected_result
+
+
+def test_custom_range_string_with_stop_returns_part_of_string_0_to_stop():
+    actual_result = custom_range(string.ascii_lowercase, stop="g")
```
Comment (Cjkjvfnby) [inline test/test_02_05.py:83]: It should be: `custom_range(string.ascii_lowercase, "g")`

File: test/test_02_05.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        "b",
+        2,
+    )
+    expected_result = [("a", 1), ("robin", "the brave"), ("d", 8)]
+    assert actual_result == expected_result
+
+
+def test_custom_range_dictionary_with_start_key_stop_key_returns_part_of_list_with_items():
+    actual_result = custom_range(
+        {
```
Comment (Cjkjvfnby) [inline test/test_02_05.py:109]: What will be the result of iteration over? 

`{
            "a": 1,
            "gallahad": "the pure",
            "robin": "the brave",
            "c": 5,
            "d": 8,
            "f": 0,
            "b": 2,
        }`
        
PS. Open the console and try it.

File: hm_1/tasks/task_02_05.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    if not some_iterable_object:
+        return "Wrong values"
+    start = 0 if start == "" else detect_index(some_iterable_object, start)
```
Comment (Cjkjvfnby) [inline hm_1/tasks/task_02_05.py:13]: What does `0` mean?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Please add a task description.

File: hm_1/tasks/task_02_05.py (hunk 4 above)
Comment (mukhaw) [reply hm_1/tasks/task_02_05.py:13]: Zéro index or some_iterable_object

File: hm_1/tasks/task_02_05.py (hunk 1 above)
Comment (mukhaw) [reply hm_1/tasks/task_02_05.py:10]: Yes, and it returns a "Wrong value"

Comment (mukhaw) [reply hm_1/tasks/task_02_05.py:10]: Or you ask about  this situation [''g','','o'] ?

File: test/test_02_05.py (hunk 3 above)
Comment (mukhaw) [reply test/test_02_05.py:109]: And it would return all dictionary, because we don't have start, stop, step

File: hm_1/tasks/task_02_05.py [hunk 5]
```diff
@@ -7,15 +7,35 @@ def detect_index(some_iterable_object, position):
     return some_iterable_object.index(position)
 
 
-def custom_range(some_iterable_object, start="", stop="", step=1):
+def detect_stop_start_step(some_iterable_object, args):
+    if not args:
+        args = 0, len(some_iterable_object), 1
+        return args
```
Comment (Cjkjvfnby) [inline hm_1/tasks/task_02_05.py:13]: Keep it simple. Same for other code.
```suggestion
        return 0, len(some_iterable_object), 1
```

File: hm_1/tasks/task_02_05.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
+def detect_index(some_iterable_object, position):
+    if isinstance(some_iterable_object, dict):
```
Comment (Cjkjvfnby) [inline hm_1/tasks/task_02_05.py:2]: When the task says iterable it means any iterable and you may use only the iteration part.  You don't need to go deeper and handle some special cases of iterable.  Please remove this check from here.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**